scripts/histmakers/w_z_gen_dists.py

```diff
@@ -96,11 +96,11 @@ def build_graph(df, dataset):
 
     if args.singleLeptonHists and isW or isZ:
         if isW:
-            df = df.Define('ptPrefsrLep', 'genlanti.pt()')
-            df = df.Define('etaPrefsrLep', 'genlanti.eta()')
-        else:
             df = df.Define('ptPrefsrLep', 'genl.pt()')
             df = df.Define('etaPrefsrLep', 'genl.eta()')
+        else:
+            df = df.Define('ptPrefsrLep', 'genlanti.pt()')
```

Comment:
For wlike this should actually depend on whether we are selecting the mu+ or mu- (ie changing sign for even vs odd events)

Reply:
For gen I don't think we care about W-like, should just plot both leptons.